ga4gh/datamodel/__init__.py

```diff
@@ -187,3 +218,21 @@ def _scanDataFiles(self, dataDir, patterns):
             atexit.register(_cleanupHtslibsMess, indexDir)
         if numDataFiles == 0:
             raise exceptions.EmptyDirException(dataDir, patterns)
+
+    def getFileHandle(self, dataFile):
+        try:
```

Comment:
the try / except pattern should only be used for exceptional conditions.  I don't think a cache miss qualifies.

Instead use the `if dataFile in self.memo_table: ... else: ...` pattern